### ISSUE TITLE
overwrite IfSourceNewer not working as expected with Azure File Share

### ISSUE THREAD

**Issue by opeongo** [Azure/azure-storage-azcopy#1942]
Possibly a duplicate of #910 

##### Note: The version is visible when running AzCopy without any argument
10.16.1

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows 10 Pro 21H2 - X64

##### Note: Please remove the SAS to avoid exposing your credentials. If you cannot remember the exact command, please retrieve it from the beginning of the log file.
I am using the following pair of commands:

Copy up to Azure file share from local file system:
```
azcopy copy C:\mylocalfolder\* https://myshare.file.core.windows.net/sct-regional/sct/?se=2023-08-31t05%3A06%3A46z&sig=-REDACTED-&sp=rwdlacupiytfx&spr=https&srt=sco&ss=bfqt&st=2022-08-30t21%3A06%3A46z&sv=2021-06-08 --include-regex myregex.* --overwrite=ifSourceNewer --preserve-smb-info --put-md5 --recursive
```

Copy down to local file system from Azure file share:
```
azcopy copy https://myshare.file.core.windows.net/sct-regional/sct/?se=2023-08-31t05%3A06%3A46z&sig=-REDACTED-&sp=rwdlacupiytfx&spr=https&srt=sco&ss=bfqt&st=2022-08-30t21%3A06%3A46z&sv=2021-06-08 C:\mylocalfolder\* --check-md5=failIfDifferent --include-regex myregex.* --overwrite=ifSourceNewer --preserve-last-modified-time --preserve-smb-info --recursive
```
### What problem was encountered?
My intended use case is to synchronize *selected* files (according to the regular expression) between my local file system and an Azure file share.  I am using a regular expression to select which files should be moved up to Azure or down to local.  I want to preserve the smb information (last modified time) because this is significant.   However, when the files are copied up to the Azure file share the last modified time is set to the **upload time** and this appears to be the source of the problem.

Repeatedly using the upload copy command with the --overwrite=ifSourceNewer **_seems_** to work ok.  If I run the upload command repeatedly the files are only uploaded the first time, and 'skipped' on subsequent invocations.  However, as mentioned the last modification time on the uploaded files is the upload time, so of course it is a newer time than the last modified time of the original files and the copy is skipped.

When I run the copy command multiple times to download the files to the local system the files are **_always_** copied on every invocation.  The files that are copied on to the local files system correctly get the original last modification time of the original files (I am guessing that the smb info is stored somewhere in the Azure file share passed along so that it can be restored when the file is copied).  Perhaps the azcopy command only compares the Azure file system last modified time (the time of the upload) to the last modified time on the local file system (the original last modified time), and it always finds that the Azure file system copy is 'newer'.

### How can we reproduce the problem in the simplest way?
Try using the copy commands listed above in a similar way.

### Have you found a mitigation/solution?
No I have not found a suitable mitigation.  These files are large and it is inconvenient that this synchronization technique does not work, and the files are always downloaded when they do not need to be.

**Comment by adreed-msft** [comment]
We are currently working on releasing a hotfix with changes for this.

**Comment by opeongo** [comment]
By the way, I had a look at the LastWriteTime property that is returned by the REST API for ?restype=directory&comp=list&include=Timestamps and I noticed that the last modification time in Azure has about micro-second accuracy, whereas the time on the local filesystem (NTFS I supposed) seems to be rounded to the second, at least according to the Java File.lastModified() API that I used to retrieve that info.  When I was comparing the remote and local time stamps I was getting inconsistent results until I started to round the Azure time to the closest second.  

I guess this is something to keep in mind when doing the IfSourceNewer comparison.
